apps/server/src/modules/oauth/service/oauth.service.ts

```diff
@@ -0,0 +1,159 @@
+import jwt from 'jsonwebtoken';
+import { HttpService } from '@nestjs/axios';
+import JwksRsa from 'jwks-rsa';
+import QueryString from 'qs';
+import { lastValueFrom } from 'rxjs';
+import { Injectable } from '@nestjs/common/decorators/core/injectable.decorator';
+import { System, User } from '@shared/domain';
+import { Inject } from '@nestjs/common';
+import { Logger } from '@src/core/logger';
+import { SymetricKeyEncryptionService } from '@shared/infra/encryption';
+import { SystemRepo, UserRepo } from '@shared/repo';
+import { Configuration } from '@hpi-schul-cloud/commons';
+import { TokenRequestPayloadMapper } from '../mapper/token-request-payload.mapper';
+import { TokenRequestPayload } from '../controller/dto/token-request.payload';
+import { OAuthSSOError } from '../error/oauth-sso.error';
+import { OauthTokenResponse } from '../controller/dto/oauth-token.response';
+import { FeathersJwtProvider } from '../../authorization';
+import { IservOAuthService } from './iserv-oauth.service';
+import { IJWT } from '../interface/jwt.base.interface';
+import { OAuthResponse } from './dto/oauth.response';
+import { AuthorizationParams } from '../controller/dto/authorization.params';
+
+@Injectable()
+export class OAuthService {
+	constructor(
+		private readonly userRepo: UserRepo,
+		private readonly systemRepo: SystemRepo,
+		private readonly jwtService: FeathersJwtProvider,
+		private httpService: HttpService,
+		@Inject('OAuthEncryptionService') private readonly oAuthEncryptionService: SymetricKeyEncryptionService,
+		private iservOauthService: IservOAuthService,
+		private logger: Logger
+	) {
+		this.logger.setContext(OAuthService.name);
+	}
+
+	/**
+	 * @query query input that has either a code or an error
+	 * @return authorization code or throws an error
+	 */
+	checkAuthorizationCode(query: AuthorizationParams): string {
+		if (query.code) return query.code;
+		let errorCode = 'sso_auth_code_step';
+		if (query.error) {
+			errorCode = `sso_oauth_${query.error}`;
+			this.logger.error(`SSO Oauth authorization code request return with an error: ${query.error}`);
```

Comment:
Should be outside the if statement. we want to log every error

Reply:
We already log every error in try catch block of the oauth.service.ts